domain/src/Plugin/Condition/Domain.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\domain\Plugin\Condition\Domain.
+ */
+
+namespace Drupal\domain\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\Context\ContextDefinition;
+
+/**
+ * Provides a 'Domain' condition.
+ *
+ * @Condition(
+ *   id = "domain",
+ *   label = @Translation("Domain")
+ * )
+ *
+ */
+class Domain extends ConditionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+
+    $options = array();
+
+    foreach (domain_load_and_sort() as $key => $value) {
+      $options[$key] = $value->get('name');
+    }
+
+    $form['domains'] = array(
+      '#type' => 'checkboxes',
```

Comment:
From a UI point of view, checkboxes may not scale beyond 20 domains. Are there better options here?